### ISSUE TITLE
"chameleon.tal.RepeatDict": inadequate base type

### ISSUE THREAD

**Issue by d-maurer** [malthe/chameleon#304]
`chameleon.tal.RepeatDict` derives from `dict`, but only `__setitem__` and `__getitem__` work as expected. The following transscript shows (as an example), that `keys` does not work:
```
>>> from chameleon.tal import RepeatDict
>>> d={}
>>> rd = RepeatDict(d)
>>> rd["a"] = "A"
>>> rd.keys()
dict_keys([])
>>> d.keys()
dict_keys(['a'])
```

I recommend that `RepeatDict` derives from `object` instead of `dict`, because the inherited `dict` methods do not work (and are not needed).

"https://github.com/zopefoundation/Zope/pull/807#issuecomment-603153348"
and "https://github.com/zopefoundation/Zope/pull/807#issuecomment-603163840"
describe problems caused by the inadequate base class.

**Comment by malthe** [comment]
I agree – it should just define a slot for its dictionary and not pretend to be a dict.

**Comment by malthe** [comment]
Fixed in 4acf9634db8409f404a0bccd855bb956f93446cb.
